demos/projects/ESPRESSIF/esp32/config/azure_iot_config.h

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) Microsoft Corporation.
+   Licensed under the MIT License. */
+
+#ifndef AZURE_IOT_CONFIG_H
+#define AZURE_IOT_CONFIG_H
+
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for AzureIoT middleware.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for AzureIoT middleware.
+ */
+
+#include "logging_levels.h"
```

Comment:
how about using esp32 logging infra?

[https://github.com/espressif/esp-idf/blob/166c30e7b2ed1dcaae56179329540a862915208a/components/log/include/esp_log.h](https://github.com/espressif/esp-idf/blob/166c30e7b2ed1dcaae56179329540a862915208a/components/log/include/esp_log.h)

This way you do not need logging_level.h or logging_stack.h

Reply:
That's a good idea. Although my attempts to use esp logging didn't go well.
I propose we go with the logging system we have in place (which works as of now), and later I'll make a change after I meet with someone from Espressif to help plug this up. Would that work for you?

Reply:
how about doing this:

#define AZLogError( message )    esp_log_write(ESP_LOG_ERROR, "azureiot", message );
#define AZLogInfo( message )     esp_log_write(ESP_LOG_INFO, "azureiot", message );
// #define AZLogWarn( message )     esp_log_write(ESP_LOG_WARN, "azureiot", message );
// #define AZLogDebug( message )    esp_log_write(ESP_LOG_DEBUG, "azureiot", message );


uncomment line if you need more logging.

Reply:
Done. Thanks for the help, Himanshu.